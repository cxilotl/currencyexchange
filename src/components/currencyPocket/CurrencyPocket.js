import React from "react";
import PropTypes from 'prop-types';
import cssStyles from './CurrencyPocket.module.scss';
import CurrencyInputForm from '../currencyInputForm/CurrencyInputForm';
import AccountBalance from '../accountBalance/AccountBalance';

const CurrencyPocket = ({ currency, currencySymbol, balance, onChangeAmount }) => {
  return(
    <section className={ cssStyles.layout }>
      <CurrencyInputForm
        currency={ currency }
        onChange={ onChangeAmount }
      />
      <AccountBalance
        currencySymbol={ currencySymbol }
        balance = { balance }
      />
    </section>
  );
};

CurrencyPocket.propTypes = {
  currency: PropTypes.string.isRequired,
  currencySymbol: PropTypes.string.isRequired,
  balance: PropTypes.number.isRequired,
  onChangeAmount: PropTypes.func
};

export default CurrencyPocket;