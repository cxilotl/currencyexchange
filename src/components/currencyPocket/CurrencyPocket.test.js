import React from 'react';
import renderer from 'react-test-renderer';
import CurrencyPocket from "./CurrencyPocket";

describe('CurrencyPocket', () => {
  const currency = 'gbp';
  const symbol = 'Â£';
  const balance = 58.33;
  const onInput = jest.fn();

  it('Snapshot', () => {
    const component = renderer.create(
      <CurrencyPocket
        currency={ currency }
        currencySymbol={ symbol }
        balance={ balance }
        onChangeAmount={ onInput }
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});