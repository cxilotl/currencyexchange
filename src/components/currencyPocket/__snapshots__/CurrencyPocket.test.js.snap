// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyPocket Snapshot 1`] = `
<section
  className="layout"
>
  <form
    className="form"
  >
    <label
      className="label"
      htmlFor="currentCurrency"
    >
      GBP
      <input
        className="input"
        data-testid="currency-input"
        id="currentCurrency"
        name="currentCurrency"
        onChange={[Function]}
        pattern="/^-?[0-9]+(\\\\.[0-9][0-9]?)?$/g"
        placeholder="0.00"
        type="number"
      />
    </label>
  </form>
  <div
    className="layout"
  >
    <p
      className="text"
    >
      You have 
      <span
        data-testid="account-balance"
      >
        Â£
        58.33
      </span>
    </p>
  </div>
</section>
`;
